assets/js/blocks/cart-checkout/payment-methods/saved-payment-method-options.js

```diff
@@ -25,139 +22,91 @@ import { getPaymentMethods } from '@woocommerce/blocks-registry';
-				payment_method: method.gateway,
-				[ savedTokenKey ]: token + '',
-				isSavedToken: true,
-			} );
-		},
-	};
+		method.brand,
+		method.last4,
+		expires
+	);
 };
 
 /**
  * Returns the option object for any non specific saved payment method.
  *
  * @param {CustomerPaymentMethod} savedPaymentMethod
- * @param {function(string):void} setActivePaymentMethod
- * @param {PaymentStatusDispatch} setPaymentStatus
- *
- * @return {Object} An option objects to use for RadioControl.
+ * @return {string} label
  */
-const getDefaultPaymentMethodOptions = (
-	{ method, tokenId },
-	setActivePaymentMethod,
-	setPaymentStatus
-) => {
-	return {
-		value: tokenId + '',
-		label: sprintf(
-			/* translators: %s is the name of the payment method gateway. */
-			__( 'Saved token for %s', 'woo-gutenberg-products-block' ),
-			method.gateway
-		),
-		name: `wc-saved-payment-method-token-${ tokenId }`,
-		onChange: ( token ) => {
-			const savedTokenKey = `wc-${ method.gateway }-payment-token`;
-			setActivePaymentMethod( method.gateway );
-			setPaymentStatus().started( {
-				payment_method: method.gateway,
-				[ savedTokenKey ]: token + '',
-				isSavedToken: true,
-			} );
-		},
-	};
+const getDefaultLabel = ( { method } ) => {
+	return sprintf(
+		/* translators: %s is the name of the payment method gateway. */
+		__( 'Saved token for %s', 'woo-gutenberg-products-block' ),
+		method.gateway
+	);
 };
 
 const SavedPaymentMethodOptions = () => {
 	const {
-		setPaymentStatus,
 		customerPaymentMethods,
 		activePaymentMethod,
 		setActivePaymentMethod,
 		activeSavedToken,
-		setActiveSavedToken,
 	} = usePaymentMethodDataContext();
-	const standardMethods = getPaymentMethods();
 	const { paymentMethods } = usePaymentMethods();
 	const paymentMethodInterface = usePaymentMethodInterface();
+	const { noticeContexts } = useEmitResponse();
+	const { removeNotice } = useStoreNotices();
+	const { dispatchCheckoutEvent } = useStoreEvents();
 
-	/**
-	 * @type      {Object} Options
-	 * @property  {Array}  current  The current options on the type.
-	 */
-	const currentOptions = useRef( [] );
-
-	const updateToken = useCallback(
-		( token ) => {
-			setActiveSavedToken( token );
-		},
-		[ setActiveSavedToken ]
-	);
-
-	useEffect( () => {
+	const options = useMemo( () => {
 		const types = Object.keys( customerPaymentMethods );
-		const options = types
+		return types
 			.flatMap( ( type ) => {
 				const typeMethods = customerPaymentMethods[ type ];
 				return typeMethods.map( ( paymentMethod ) => {
-					const option =
-						type === 'cc' || type === 'echeck'
-							? getCcOrEcheckPaymentMethodOption(
-									paymentMethod,
-									setActivePaymentMethod,
-									setPaymentStatus
-							  )
-							: getDefaultPaymentMethodOptions(
-									paymentMethod,
-									setActivePaymentMethod,
-									setPaymentStatus
-							  );
-					if (
-						! activePaymentMethod &&
-						paymentMethod.is_default &&
-						activeSavedToken === ''
```

Comment:
Not sure why we were handling defaults here. I moved it to context with the other default setter.